src/value/value.rs

```diff
@@ -1,14 +1,19 @@
 use std::str::Split;
-use std::collections::BTreeMap;
-
 use serde::Serialize;
+#[cfg(not(feature = "preserve_order"))]
+use std::collections::BTreeMap;
 
 use crate::value::{Tag, ValueSerializer};
 use crate::error::{Error, Actual};
 
 /// An alias to the type of map used in [`Value::Dict`].
+#[cfg(not(feature = "preserve_order"))]
 pub type Map<K, V> = BTreeMap<K, V>;
 
+/// An alias to the type of map used in [`Value::Dict`].
+#[cfg(feature = "preserve_order")]
+pub type Map<K, V> = indexmap::IndexMap<K, V>;
```

Comment:
We cannot do this: features [_must_ be additive](https://doc.rust-lang.org/cargo/reference/features.html?highlight=additive#feature-unification). This breaks that.